core/test-src/org/pentaho/di/core/ConstTest.java

```diff
@@ -1474,4 +1540,464 @@ public static void assertEquals( String msg, Object expected, Object actual ) {
       Assert.assertEquals( msg, expected, actual );
     }
   }
+
+  @Test
+  public void testToInt() {
+    assertEquals( 123, Const.toInt( "123", -12 ) );
+    assertEquals( -12, Const.toInt( "123f", -12 ) );
+  }
+
+  @Test
+  public void testToLong() {
+    assertEquals( 1447252914241L, Const.toLong( "1447252914241", -12 ) );
+    assertEquals( -1447252914241L, Const.toLong( "1447252914241L", -1447252914241L ) );
+  }
+
+  @Test
+  public void testToDouble() {
+    Assert.assertEquals( 123.45, Const.toDouble( "123.45", -12.34 ), 1e-15 );
+    Assert.assertEquals( -12.34, Const.toDouble( "123asd", -12.34 ), 1e-15 );
+  }
+
+  @Test
+  public void testToDate() {
+    final Date expected = new Date( 1447252914241L );
+    final Date def = new Date( 12312312123L );
+
+    assertEquals( expected, Const.toDate( "2015/11/11 14:41:54.241", def ) );
+    assertEquals( def, Const.toDate( "invalid date", def ) );
+  }
+
+  @Test
+  public void testRightPad() {
+    final String s = "Pad me baby one more time";
+
+    assertEquals( "     ", Const.rightPad( (String) null, 5 ) );
+    assertEquals( "Pad", Const.rightPad( s, 3 ) );
+
+    final StringBuffer sb = new StringBuffer( s );
+    assertEquals( s + "   ", Const.rightPad( sb, 28 ) );
+    assertEquals( "Pad me baby", Const.rightPad( sb, 11 ) );
+  }
+
+  @Test
+  public void testReplace() {
+    final String source = "A journey of a thousand miles never begins";
+    assertEquals( "A journey of a thousand miles begins with a single step", Const.replace( source, "never begins",
+        "begins with a single step" ) );
+    assertEquals( source, Const.replace( source, "evil", "good" ) );
+    assertEquals( "short", Const.replace( "short", "long pattern", "replacement" ) );
+    assertEquals( "", Const.replace( "", "anything", "something" ) );
+  }
+
+  @Test
+  public void testRepl() {
+    String source = "A journey of a thousand miles never begins";
+    StringBuffer sb = new StringBuffer( source );
+    Const.repl( sb, "never begins", "begins with a single step" );
+    assertEquals( "A journey of a thousand miles begins with a single step", sb.toString() );
+    sb = new StringBuffer( source );
+    Const.repl( sb, "evil", "good" );
+    assertEquals( source, sb.toString() );
+    sb = new StringBuffer( "short" );
+    Const.repl( sb, "long pattern", "replacement" );
+    assertEquals( "short", sb.toString() );
+    sb = new StringBuffer( "" );
+    Const.repl( sb, "anything", "something" );
+    assertEquals( "", sb.toString() );
+  }
+
```

Comment:
@mkambol I'm not, remember that tiny and very functional OS, I've put great expectations on it, but unfortunately it's now far from what I expected and almost dead (checked several years ago and development efforts put in it was near 0).